### ISSUE TITLE
Momentum Scheduling

### ISSUE THREAD

**Issue by stalhabukhari** [Kamnitsask/deepmedic#175]
Hi!

I am trying to understand the scheme used for scheduling momentum and came across [Line 257](https://github.com/Kamnitsask/deepmedic/blob/master/deepmedic/neuralnet/trainer.py#L257):

    `y_min = self._init_lr_tfv`

Within the context of the file, **self._init_lr_tfv** appears to be the initial learning rate variable defined in Tensorflow. If so, then **y_min** should instead correspond to the initial momentum, defined in code as **self._init_mom_tfv**.

If this observation is correct then the last term of momentum update formula in [Line 260](https://github.com/Kamnitsask/deepmedic/blob/master/deepmedic/neuralnet/trainer.py#L260) should be **y_min** instead of **y_max** (I infer that we are linearly interpolating from **y_min** to **y_max**).

Kindly provide some insight into the scheme, any reference material from where this scheme is taken would be very helpful.

**Comment by Kamnitsask** [comment]
Ha! Very nice eye for detail you have there.

Yes, it should be y_min = self._init_mom_tfv there.

Note this part of the code is not used, as it's only used if a specific learning-rate-schedule was selected, "expon"(-ential decay). Which never worked well for me so I never enabled it in the trainConfigs. Now this issue may explain why it never worked as good as other schedules :) (although I think it didnt work for me even back when deepmedic was in theano days, where I dont know if this bug was there, but anyway). If you dont care much about training schedules, I d still suggest the pre-defined or "poly", which has given ok results for me in a few databases.

There is no reference for this "expon" schedule, it is simply something I made to try among others.

Thanks a lot for spotting this. Fixed in commit 997dd3c now.
